patches/spirv/0001-Fix-debug-info-of-work-item-builtin-translation-745.patch

```diff
@@ -0,0 +1,118 @@
+From 4474baa8d1a4f4b5ba9684068ff00bfefbf541eb Mon Sep 17 00:00:00 2001
+From: Wenju He <wenju.he@intel.com>
+Date: Fri, 9 Oct 2020 14:21:09 +0800
+Subject: [PATCH] Fix debug info of work-item builtin translation (#745)
+
+debug info of work-item builtins are lost in both llvm IR -> spirv and
+spirv -> llvm IR translations. See #744
+---
+ lib/SPIRV/OCL20ToSPIRV.cpp              |  7 ++-
+ lib/SPIRV/SPIRVReader.cpp               |  2 +
+ test/DebugInfo/builtin-get-global-id.ll | 60 +++++++++++++++++++++++++
+ 3 files changed, 68 insertions(+), 1 deletion(-)
+ create mode 100644 test/DebugInfo/builtin-get-global-id.ll
+
+diff --git a/lib/SPIRV/OCL20ToSPIRV.cpp b/lib/SPIRV/OCL20ToSPIRV.cpp
+index 1262c48..d38392b 100644
+--- a/lib/SPIRV/OCL20ToSPIRV.cpp
++++ b/lib/SPIRV/OCL20ToSPIRV.cpp
+@@ -1297,11 +1297,16 @@ void OCL20ToSPIRV::transWorkItemBuiltinsToVariables() {
+     for (auto UI = I.user_begin(), UE = I.user_end(); UI != UE; ++UI) {
+       auto CI = dyn_cast<CallInst>(*UI);
+       assert(CI && "invalid instruction");
+-      Value *NewValue = new LoadInst(BV, "", CI);
++      const DebugLoc &DLoc = CI->getDebugLoc();
++      Instruction *NewValue = new LoadInst(BV, "", CI);
++      if (DLoc)
```

Comment:
DLoc is not a pointer. why use it directly here?   DLoc.get() ?   
And I think this check is not necessary.

Reply:
DebugLoc has operator bool () const

Reply:
Done